pkg/registry/core/shoot/storage/binding.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gardener/gardener/pkg/apis/core"
+	"github.com/gardener/gardener/pkg/apis/core/validation"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apiserver/pkg/registry/rest"
+	"k8s.io/apiserver/pkg/storage"
+	"k8s.io/apiserver/pkg/util/dryrun"
+)
+
+// BindingREST implements a RESTStorage for shoots/binding.
+type BindingREST struct {
+	shootStorage *REST
+}
+
+var _ = rest.NamedCreater(&BindingREST{})
+
+// New returns and instance of Binding
+func (r *BindingREST) New() runtime.Object {
+	return &core.Binding{}
+}
+
+// Create binds a shoot to a seed
+func (r *BindingREST) Create(ctx context.Context, name string, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (out runtime.Object, err error) {
+	binding, ok := obj.(*core.Binding)
+	if !ok {
+		return nil, errors.NewBadRequest(fmt.Sprintf("not a Binding object: %#v", obj))
+	}
+
+	if name != binding.Name {
+		return nil, errors.NewBadRequest(fmt.Sprintf("name in URL does not match name in Binding object: %v,%v", name, binding.Name))
+	}
+
+	if createValidation != nil {
+		if err := createValidation(ctx, binding.DeepCopyObject()); err != nil {
+			return nil, err
+		}
+	}
+
+	if errs := validation.ValidateBinding(binding); len(errs) != 0 {
+		return nil, errs.ToAggregate()
+	}
+
+	shootObj, err := r.shootStorage.Get(ctx, name, &metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	shoot, ok := shootObj.(*core.Shoot)
+	if !ok {
+		return nil, errors.NewInternalError(fmt.Errorf("cannot convert to *core.Shoot object - got type %T", shootObj))
+	}
+
+	if err := r.assignShoot(ctx, binding.UID, binding.ResourceVersion, shoot.Name, binding.Target.Name, dryrun.IsDryRun(options.DryRun)); err != nil {
```

Comment:
There is the `SeedChange` feature gate. When the feature gate is set to false, it should not be allowed to change the `.spec.seedName` field. This validation is also skipped/violated with this PR. With `SeedChange=false` I was able to change the seedName field.

https://github.com/gardener/gardener/blob/8f819ffab82898c65be19a4376aeec3b5627c3ba/pkg/apis/core/validation/shoot.go#L244-L247

cc @plkokanov

Reply:
Yes yes, I am taking care of it as mentioned in the notes for reviewer. I shouldn't have marked the PR ready for review, sorry for that. Just wanted to run the tests in prow.